client/landing/stepper/declarative-flow/internals/steps-repository/unified-domains/index.tsx

```diff
@@ -80,33 +80,32 @@ const RenderDomainsStepConnect = connect(
 	}
 )( withCartKey( withShoppingCart( localize( RenderDomainsStep ) ) ) );
 
+/**
+ * The domains step has a quirk where it calls `submitSignupStep` then synchronously calls `goToNextStep` after it.
+ * This doesn't give `setStepState` a chance to update and the data is not passed to `submit`.
+ */
+let mostRecentState: ProvidedDependencies = {};
+
 export default function DomainsStep( props: StepProps ) {
 	const [ stepState, setStepState ] =
 		useStepPersistedState< ProvidedDependencies >( 'domains-step' );
 	const managedSiteFlowProps = useIsManagedSiteFlowProps();
 
-	const mostRecentStateRef = useRef< ProvidedDependencies | undefined >( undefined );
-
-	const updateSignupStepState = useCallback(
-		( state: ProvidedDependencies, providedDependencies: ProvidedDependencies ) => {
-			setStepState(
-				( mostRecentStateRef.current = { ...stepState, ...providedDependencies, ...state } )
-			);
-		},
-		[ stepState, setStepState ]
-	);
-
 	return (
 		<CalypsoShoppingCartProvider>
 			<RenderDomainsStepConnect
 				{ ...props }
 				{ ...managedSiteFlowProps }
 				page={ ( url: string ) => window.location.assign( url ) }
-				saveSignupStep={ updateSignupStepState }
-				submitSignupStep={ updateSignupStepState }
+				saveSignupStep={ ( step: ProvidedDependencies ) => {
+					setStepState( ( mostRecentState = { ...stepState, ...step } ) );
+				} }
+				submitSignupStep={ ( _, step: ProvidedDependencies ) => {
```

Comment:
the `_` is the main change here. To pick up the second param instead of the first.